### DISCUSSION TITLE
Remote Rendering with pvserver

### DISCUSSION THREAD

**Discussion by jwindgassen** [Kitware/trame#145] [Q&A] [answered]
We are interested in using trame ([paraview-visualizer](https://github.com/Kitware/paraview-visualizer) in our JupyterLab to be exact) to be able to interact with and visualize a running pvserver on another machine. 

From my understanding, trame will launch its own instance of ParaView. We were wondering if you could instead use trame and connect to an already running instance, similar to using ParaView's `Connect`-Feature.
I already thought about launching the application instead of pvserver on the remove machine and then just accessing the URL. But we also wanted to use that server as an endpoint for Catalyst2 and I doubt, that this would be possible then.

I just started using this framework and am curious, how and if interfacing with other ParaView instances is even possible with trame.

**Comment by jourdain** [comment, marked as answer]
Hi @jwindgassen,

Thanks for reaching out. So that capability is already supported with some caveat for your use-case.

First of all, you can run a trame app within pvpython and ask `paraview.simple` to `Connect` to a remote `pvserver`. It is probably not currently exposed on paraview-visualizer but it will only takes few minutes to add such features unless you want fancy dynamic UI for triggering such connect. 

Then for having a ParaView based trame app within Jupyter is a different story. For that part you have 2 options.

The first option is to have the trame server running yet on another server and just having an iFrame in a Jupyter cell. But at that point I'm not sure why you will need that trame app within Jupyter.

The second option is to have ParaView (client/pvpython) running within the Jupyter kernel. Either by using the conda-forge version of paraview or by running Jupyter from within pvpython+venv. The limitation of using the ParaView binary to run Jupyter within is that our binaries don't come with SSL which is required by tornado/jupyter. But if you figure out a path that is working for you on that end, you will be able to access your ParaView data from within your other Jupyter cells (assuming you fetch and call the proper methods to get what you need that live remotely on pvserver).

Side note: paraview-visualizer is still in alpha but eventually once I manage to spend some time on it, it should become pretty good. But any help on it is welcome (funding or man power).

HTH,

Seb

**Reply by jwindgassen** [reply]
So the way I am currently trying to integrate trame into out JupyterLab is using a [JupyterServerProxy](https://jupyter-server-proxy.readthedocs.io/en/latest/index.html) because we used that way for [Xpra](https://github.com/FZJ-JSC/jupyter-xprahtml5-proxy/tree/main/jupyter_xprahtml5_proxy) and [VSCode](https://github.com/FZJ-JSC/jupyter-openvscodeserver-proxy) (although the latter is not yet fully working on our system) and it is a very convenient way of accessing a hosted website. 

Unfortunately, there seems to be a problem with trame using WebSockets. Opening the Website works fine, but the browser is unable to connect to the WebSocket of the Server over the HTTPS connection JupyterLab uses:
```
Mixed Content: The page at '<lab-url>/trame/index.html' was loaded over HTTPS, but attempted to connect to the insecure WebSocket endpoint 'ws://<lab-url>/trame/ws'. This request has been blocked; this endpoint must be available over WSS

DOMException: Failed to construct 'WebSocket': An insecure WebSocket connection may not be initiated from a page loaded over HTTPS.
```
We integrated ParaViewWeb into JupyterLab [in the past](https://gitlab.jsc.fz-juelich.de/jupyter4jsc/j4j_extras/pvlink/-/blob/master/pvlink/remoterenderer.py#L206) and circumvented that problem by manually setting the `sessionURL` in wslink to use wss instead of ws, but I was unable to find a suiting replacement here.

I also found [the start script in the docker container](../blob/master/docker/scripts/start.sh), which mentions how to switch to wss, but I could not find the correlating option in trame.

**Reply by jourdain** [reply]
I think that issue was fixed recently and you might just need to do a `pip install -U trame-client` to get things working (+clearing your browser cache).

**Reply by jwindgassen** [reply]
Yes, that indeed fixed the issue, tyvm. Unfortunately, that didn't fix all the problems. I moved it to #159 for now, as it seemed the more appropriate place!

**Reply by jwindgassen** [reply]
FYI: **v0.1 of the Jupyter trame Proxy** can be found [here!](https://github.com/jwindgassen/jupyter-trame-proxy)

Another question regarding this Proxy: trame mentions the possibility to create a custom launcher on multi-user setup, which is the case for our JupyterLab. Currently, we don't feel ready to release the proxy to production, because we fear the insecurity from the unprotected port.

I found #32, which mentions the Cookiecutter for trame v2, but I am unsure if that works with paraview-visualizer. Is there any equivalent to the launcher / possibility of a protected webserver on multi-user systems already in paraview-visualizer?

**Reply by jourdain** [reply]
Thanks for sharing your work for integrating trame/visualizer as an extension to Jupyter. 

Our [cookie-cutter](https://github.com/Kitware/trame-cookiecutter/blob/master/%7B%7Bcookiecutter.package_name%7D%7D/setup.cfg#L52-L53) try to enable some of what you are doing by default. It might be nice to improve the generic path (cookie-cutter) following some of the step you did and maybe even provide that within Visualizer itself. 

From my understanding if you get a new port each time you run that command, that means you already have a "launcher" to support multi-clients. I guess the only remaining part is to prevent "others" from connecting to a port to get the app of someone else. To prevent that you need to do the following:
1. Add a generated token in the command line `--authKey {token-secret-value}`
2. Then connect with the URL `...?secret={token-secret-value}`

HTH,

Seb

**Reply by jwindgassen** [reply]
I understand. So the mentioned launcher is just for preventing, that multiple user try to run trame on the same port, which would not work, right? If that is so, you are correct: If you use "{port}" anywhere, JupyterServerProxy will automatically find an open port and replace the pattern with the value. So there shouldn't be a problem with blocked ports (hopefully).

I also added the authKey to make sure the data can't be seen by anyone (Although I used `--authKeyFile` instead of `--authKey`, I already explained how and why in Kitware/wslink#124).

Does trame also allow specifying how the underlying ParaView is launched? We have multiple systems with different configurations (hardware and software) and atm use some custom bash-scripts to make sure, that ParaView is running optimally on all systems. Is there by chance a way how we can integrate such a script?

**Reply by jourdain** [reply]
So the mention of the launcher info at the start is more about letting the human who started the process that he is not starting a server for multiple users (like you could expect from any other web server). Within the cookiecutter, we provide a path to automate multi-users via a docker image where an apache front-end handle the routing to each started process internal port via url mapping like your proxy is doing using Jupyter infrastructure. The built-in launcher expect a POST request for starting a new session for the given user, then that user get redirected (just the ws) to get its app.

Thanks for the contribution of `--authKeyFile`.

Trame in itself is not really concerned on how its python environment get setup. It is more the job of the "launcher". Our built-in version (the one that expect the POST request `python -m wslink.launcher ...`) give you some freedom to adjust/configure the command line that should be executed to start the trame server. At a high level you group command lines into "application" and you can have parameters be provided from the URL at startup (which can use your shell script to streamline things even more). I'm not sure how it will work with Jupyter though. Maybe I'm not fully grasping what you are asking on `Does trame also allow specifying how the underlying ParaView is launched?`.

HTH,

Seb

**Reply by jwindgassen** [reply]
Correct me if I'm wrong, but from my understanding, trame will launch a local instance of ParaView, *connect to it * (in some way) and then render the data in ParaView and display the result in the browser. (For a very simplified overview)

That's what I assumed because when starting pv-visualizer on a Desktop machine, there will be another unresponsive window named *ParaView*.

If that is the case, I was wondering where and how the process with ParaView is being started. I couldn't find any corresponding code in the packages.

I am, unfortunately, not experienced with the ParaView internals, so this is all a bit handwavy.

**Reply by jourdain** [reply]
I think I start seeing what you are talking about. 

So ParaView has several execution mode:
1. single process (client+dataserver+renderserver)
2. client/server (client <-> pvserver=dataserver+renderserver)

Usually we run trame in the usecase 1. Meaning that everything is local and trame is running within ParaView.

If you are talking about 2, then you still need trame to run within ParaView as a client and then use the ParaView API to connect to that remote pvserver `paraview.simple.Connect(...)`.

Ideally we want to avoid using client+pvserver. At that point it is better to run pvbatch (for mpi) or pvpython with your trame app directly where you were planning to run pvserver. 

I'm not sure that I'm helping here, but in client/pvserver, trame in itself don't do anything about the pvserver process. It is up to you to code what ever logic you are looking for in python.

**Reply by jourdain** [reply]
I guess what I'm trying to say is that the trame process that you start is your ParaView server already (or it should be).

**Reply by jwindgassen** [reply]
Ok. So no matter what execution mode we are choosing, there is always a Process that is running ParaView on the machine of the user/client, right?

But if you say
> everything is local and trame is running within ParaView.
> the trame process that you start is your ParaView server already.

where does this connection between these 2 happen? You start trame via python, but at some point, python must start the ParaView that will contain the trame server. 
E.g.: `os.system('paraview')` would startup ParaView. But trame will probably use another way so that the communication between them can be established.

Where does this part happen? I suspected it is somewhere in wslink or paraview.simple, just because I couldn't find any notion of that process anywhere within trame.

I am very sorry, that I need the answer in such detail, but I need that insight for some upcoming projects we want to do. On the bright side, we have big plans and ideas for the upcoming months that will involve trame on our HPC systems :)
Thank you very much for your support!

~Jonathan

P.S.: We will add an entry to #19 once we release the proxy to procution in the next few weeks

**Reply by jourdain** [reply]
So trame is already your paraview, therefore I don't understand why you will need trame to start yet another paraview. You should strive to do case 1, not 2. And if you are doing 2, you should think why it can not do 1.

Also paraview-client/pvpython/pvbatch == trame/server. 

I guess you need to clarify what runs where and what you are trying to do because I still don't understand.

**Reply by jourdain** [reply]
My understanding is that the user/client is not running ParaView but only a browser.

**Reply by jwindgassen** [reply]
I am not talking about any specific usecase, I am just trying to understand how the backend of trame is working. But for the sake of argument, you can assume we are in case 1 for now.

I don't need trame to launch another paraview, I thought this is what it was already doing in the background when you run it. I assumed, that trame would lauch a paraview instance in the background and serve as a relay which forwards input from the user to paraview and returns the rendered image to the browser. So ParaView is the renderer, trame the interface with the web browser. 

But from this conversation it seems, that this isn't the case altogether. Instead, trame will serve as the rendering backend itself (in addition to being the server for the website). So the trame process does not "contain" a literal, real instance of Paraview, but instead is "emulating" one (or I guess generating the same output that ParaView would produce with the input data). This makes the whole thing a lot easier.

**Reply by jourdain** [reply]
I guess the somewhat accurate summary is `ParaView is the renderer, trame the interface with the web browser.`

But I think  the main miss-conception is what we call ParaView here.

1. In our case ParaView is providing our Python runtime with the ParaView/VTK library so you can script data processing and rendering.
2. Trame is a library that let you create a web front-end for driving things in and from python.
3. ParaView-Visualizer is a Python application using trame to create a web front-end to drive the ParaView library in a way that you can load, process and render 3D data interactively from a web UI. It use the Python runtime from ParaView to execute and run the trame server. That way ParaView objects get manipulated from Python by trame directly. 

So far with Jupyter, the way you are running it, is by starting a new process but you can also run that pv_visualizer within the same Python interpreter as your Jupyter kernel. When doing so, you can share object in-memory between your Jupyter Cells and ParaView data processing and visualization. But I guess that is another story altogether.

Thanks again for considering to contribute to #19.

**Reply by jwindgassen** [reply]
tyvm, I think I understand now. In my head when you talk about ParaView, I was always thinking about the big executable and never really about the python package that comes alongside it. I was not aware, that you could use it for that. Sorry for the confusion.

**Comment by jwindgassen** [comment]
> First of all, you can run a trame app within pvpython and ask `paraview.simple` to `Connect` to a remote `pvserver`. It is probably not currently exposed on paraview-visualizer but it will only takes few minutes to add such features unless you want fancy dynamic UI for triggering such connect.

As mentioned in the original post, one goal was to run trame with a pvserver. I am currently trying to do so. I am using `simple.Connect` for that. I added the following to `pv_visualizer.app.main`: 
```python
def _connect(**_):
    host = os.environ["TRAME_PVSERVER_HOST_ADDRESS"]
    connection = simple.Connect(host)
    
server.controller.on_server_ready.add(_connect)
```

But after I am connected, I am unable to load any input files nor create any of the primitive Objects:
```
# When Loading
ERROR:root:Exception raised
ERROR:root:AttributeError("'NoneType' object has no attribute 'Session'")
ERROR:root:Traceback (most recent call last):
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/wslink/backends/aiohttp/__init__.py", line 495, in onMessage
    results = await results
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/trame_server/protocol.py", line 169, in trigger
    return await asyncio.coroutine(self.server._triggers[name])(
  File "/p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/python3.9/asyncio/coroutines.py", line 124, in coro
    res = func(*args, **kw)
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/pv_visualizer/html/file_browser.py", line 262, in _update_path
    self._on_load_file(file_to_load)
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/trame_server/controller.py", line 229, in __call__
    result = self.func(*args, **kwargs)
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/pv_visualizer/app/ui/files.py", line 56, in load_file
    reader = simple.OpenDataFile(data_to_load)
  File "/p/software/juwelsbooster/stages/2022/software/ParaView/5.10.1-gpsmkl-2021b-EGL/lib64/python3.9/site-packages/paraview/simple.py", line 1184, in OpenDataFile
    session = servermanager.ActiveConnection.Session
AttributeError: 'NoneType' object has no attribute 'Session'

# Spawn Sphere
ERROR:root:Exception raised
ERROR:root:KeyError('Sphere')
ERROR:root:Traceback (most recent call last):
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/wslink/backends/aiohttp/__init__.py", line 495, in onMessage
    results = await results
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/trame_server/protocol.py", line 169, in trigger
    return await asyncio.coroutine(self.server._triggers[name])(
  File "/p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/python3.9/asyncio/coroutines.py", line 124, in coro
    res = func(*args, **kw)
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/pv_visualizer/html/filters.py", line 48, in _click
    self._click_fn(name)
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/trame_server/controller.py", line 229, in __call__
    result = self.func(*args, **kwargs)
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/pv_visualizer/app/ui/algorithms.py", line 19, in create_filter
    newProxy = simple.__dict__[name]()
KeyError: 'Sphere'

# Spawn Cone
ERROR:root:Exception raised
ERROR:root:KeyError('Cone')
ERROR:root:Traceback (most recent call last):
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/wslink/backends/aiohttp/__init__.py", line 495, in onMessage
    results = await results
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/trame_server/protocol.py", line 169, in trigger
    return await asyncio.coroutine(self.server._triggers[name])(
  File "/p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/python3.9/asyncio/coroutines.py", line 124, in coro
    res = func(*args, **kw)
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/pv_visualizer/html/filters.py", line 48, in _click
    self._click_fn(name)
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/trame_server/controller.py", line 229, in __call__
    result = self.func(*args, **kwargs)
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/pv_visualizer/app/ui/algorithms.py", line 19, in create_filter
    newProxy = simple.__dict__[name]()
KeyError: 'Cone'
```

Sometimes I also get a few dozens of
```
ERROR:root:Exception raised
ERROR:root:Exception('no view provided: 279')
ERROR:root:Traceback (most recent call last):
  File "/p/project/dems/windgassen1/jupyter/kernels/paraview-trame/lib/python3.9/site-packages/wslink/backends/aiohttp/__init__.py", line 493, in onMessage
    results = func(*args, **kwargs)
  File "/p/software/juwelsbooster/stages/2022/software/ParaView/5.10.1-gpsmkl-2021b-EGL/lib64/python3.9/site-packages/paraview/web/protocols.py", line 893, in setViewSize
    sView = self.getView(viewId)
  File "/p/software/juwelsbooster/stages/2022/software/ParaView/5.10.1-gpsmkl-2021b-EGL/lib64/python3.9/site-packages/paraview/web/protocols.py", line 132, in getView
    raise Exception("no view provided: " + str(vid))
Exception: no view provided: 279
```
which will be followed by a crash of the server and a loss of connection as soon as I click any element of the UI

Any ideas?

**Reply by jourdain** [reply]
I think this is an ordering issue where things get created in the built-in session before you get connected to the pv-server.

The connect should be the call that happen before any other paraview call.

**Reply by jwindgassen** [reply]
tyvm, that did it. Directly calling `Connect` works.
Maybe about the motivation: Our goal is to connect trame to a pvserver, which can accept a Catalyst Connection to allow visualizing on multiple nodes. 

So after I call `simple.Connect(host)` I'm calling `servermanager.ConnectToCatalyst()`, which will, again, crash the program and the server that is now connected to it:
```
Loguru caught a signal: SIGSEGV
Stack trace:
40            0x4006ce _start + 46
39      0x1548242dfd85 __libc_start_main + 229
38      0x154825b62d09 Py_BytesMain + 41
37      0x154825b838d6 Py_RunMain + 326
36      0x154825b6ff5c PyRun_SimpleStringFlags + 60
35      0x154825b70024 PyRun_StringFlags + 164
34      0x154825b73167 /p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/libpython3.9.so.1.0(+0x1b8167) [0x154825b73167]
33      0x154825b75014 /p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/libpython3.9.so.1.0(+0x1ba014) [0x154825b75014]
32      0x154825ad65cb PyEval_EvalCode + 27
31      0x154825ad6613 PyEval_EvalCodeEx + 67
30      0x154825ad6671 _PyEval_EvalCodeWithName + 81
29      0x154825ad6866 /p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/libpython3.9.so.1.0(+0x11b866) [0x154825ad6866]
28      0x154825ad97ab _PyEval_EvalFrameDefault + 7531
27      0x154825ae6d05 _PyFunction_Vectorcall + 229
26      0x154825ad6866 /p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/libpython3.9.so.1.0(+0x11b866) [0x154825ad6866]
25      0x154825ad8bcf _PyEval_EvalFrameDefault + 4495
24      0x154825ae6d05 _PyFunction_Vectorcall + 229
23      0x154825ad6866 /p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/libpython3.9.so.1.0(+0x11b866) [0x154825ad6866]
22      0x154825ad8150 _PyEval_EvalFrameDefault + 1808
21      0x154825ae6fc3 /p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/libpython3.9.so.1.0(+0x12bfc3) [0x154825ae6fc3]
20      0x154825ad8150 _PyEval_EvalFrameDefault + 1808
19      0x154825ae6fc3 /p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/libpython3.9.so.1.0(+0x12bfc3) [0x154825ae6fc3]
18      0x154825ad8bcf _PyEval_EvalFrameDefault + 4495
17      0x154825ae6fc3 /p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/libpython3.9.so.1.0(+0x12bfc3) [0x154825ae6fc3]
16      0x154825ad7fa2 _PyEval_EvalFrameDefault + 1378
15      0x154825ae6d05 _PyFunction_Vectorcall + 229
14      0x154825ad6866 /p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/libpython3.9.so.1.0(+0x11b866) [0x154825ad6866]
13      0x154825adc0ad _PyEval_EvalFrameDefault + 18029
12      0x154825ae6d05 _PyFunction_Vectorcall + 229
11      0x154825ad6866 /p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/libpython3.9.so.1.0(+0x11b866) [0x154825ad6866]
10      0x154825ad8bcf _PyEval_EvalFrameDefault + 4495
9       0x154825af02a9 /p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/libpython3.9.so.1.0(+0x1352a9) [0x154825af02a9]
8       0x154825ae6d05 _PyFunction_Vectorcall + 229
7       0x154825ad6866 /p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/libpython3.9.so.1.0(+0x11b866) [0x154825ad6866]
6       0x154825addcb3 _PyEval_EvalFrameDefault + 25203
5       0x154825af07db _PyObject_Call + 123
4       0x154825af17e8 /p/software/juwelsbooster/stages/2022/software/Python/3.9.6-GCCcore-11.2.0/lib/libpython3.9.so.1.0(+0x1367e8) [0x154825af17e8]
3       0x15480916624d /p/software/juwelsbooster/stages/2022/software/ParaView/5.10.1-gpsmkl-2021b-EGL/lib64/python3.9/site-packages/paraview/modules/vtkRemotingServerManager.so(+0xc124d) [0x15480916624d]
2       0x154808f7da96 vtkSMParaViewPipelineController::RegisterViewProxy(vtkSMProxy*, char const*) + 310
1       0x154808f8dd84 vtkSMPropertyHelper::vtkSMPropertyHelper(vtkSMProxy*, char const*, bool) + 20
0       0x154824ffecf0 /usr/lib64/libpthread.so.0(+0x12cf0) [0x154824ffecf0]
(   4.947s) [paraview        ]                       :0     FATL| Signal: SIGSEGV
```

The pvserver will show the folloing Error:
```
(   4.669s) [pvserver.0      ]vtkSocketCommunicator.c:781    ERR| vtkSocketCommunicator (0x5aa8650): Could not receive tag. 1
(   4.669s) [pvserver.0      ]vtkTCPNetworkAccessMana:296    ERR| vtkTCPNetworkAccessManager (0x55dccc0): Some error in socket processing.
```

I tested calling `Connect` and `ConnectToCatalyst` over a pvpython console beforehand, which worked without any errors.

**Reply by jourdain** [reply]
It is hard to tell, but it seems that it crash when you are creating a view. If you are connecting to catalyst you may need to get the view rather than creating a new one? (To be honest, I will need to really test/experiment to provide meaningful information at that point.) 

If you are really getting stuck you may want to reach out to Kitware in France so they can investigate how to properly achieve those actions.

**Reply by jourdain** [reply]
FYI @mwestphal

**Reply by mwestphal** [reply]
HI @jwindgassen, 

Indeed, deeper investigation is needed in order to understand what is going on.
We will reach out if that's ok.

Best,

**Reply by jwindgassen** [reply]
tyvm. Let me know if you need any further information regarding the setup or our system, I would be happy to provide!

**Reply by francois-kitware** [reply]
For the record, the crash have been fixed in ParaView. See : https://gitlab.kitware.com/paraview/paraview/-/commit/4c00f99a106cc5c29b7b55fe85348eadea50043d
